### PR TITLE
Modify Liberal Studies Checker

### PR DESCRIPTION
**Pull request by mirandayu131**
Modifies the checker for the liberal studies requirement to check whether the given course’s `crseId` is in the other yes list JSON. Additionally, it checks whether `distribution` is included in the categories listed under the other yes list for the given course. Currently, we determine whether it is classified under category `distribution` by checking if `course.catalogDistr?.includes(distribution)` is True.

Test Plan
- Test whether other Yes List Courses from https://apps.engineering.cornell.edu/liberalstudies/ApprovedLiberalStudies.pdf satisfy the Liberal Studies requirement on the server

### CODE REVIEW

**Review by zachary0kent — COMMENTED** [review]
Awesome work, everything behaves as expected! I'll approve once all the previous liberal studies PRs are merged :)

**Comment by noschiff** [comment]
Amazing work, Miranda! Super excited to release this! 🚀

File: src/requirements/checkers.ts [hunk 1]
```diff
@@ -104,6 +106,22 @@ export const courseIsFWS = (course: Course): boolean =>
...
+/**
+ * Call this function to check whether distribution is included in the course
+ * or if the course is on the Other Yes List.
+ *
+ * @param course course object with useful information retrived from Cornell courses API.
+ * @returns if the distribution is included
+ */
+export const hasCategory = (course: Course, distribution: string) => {
+  const otherYesCourse = otherYesCourses.find((x: CourseWithId) => x.crseId === course.crseId);
+  // console.log(otherYesCourse?.categories?.includes(distribution));
```
Comment (zachary0kent) [inline src/requirements/checkers.ts:118]:
```suggestion
```

File: src/requirements/checkers.ts [hunk 2]
```diff
@@ -104,6 +106,22 @@ export const courseIsFWS = (course: Course): boolean =>
...
+  return (
+    course.catalogDistr?.includes(distribution) ||
+    otherYesCourse?.categories?.includes(distribution)
```
Comment (zachary0kent) [inline src/requirements/checkers.ts:121]: Is the optional chaining after `categories` necessary--will the property `categories` every be absent on `otherYesCourse`, if it exists?

**Review by zachary0kent — CHANGES_REQUESTED** [review]

**Review by zachary0kent — APPROVED** [review]
Looks good!

**Pull request merged**